src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/service/FlowCacheService.java

```diff
@@ -91,15 +91,15 @@ public void processFlowLatencyCheck() {
 
     private void checkFlowLatency(String flowId, FlowState flowState) {
         long currentTimeMillis = clock.millis();
-        if (currentTimeMillis > flowState.getForwardPathLatency().getTimestamp() + flowRttStatsExpirationTime) {
+        if (flowState.getForwardPathLatency().isExpired(currentTimeMillis, flowRttStatsExpirationTime)) {
```

Comment:
As for me `isExpired` should be part of this service, because only it have "knowledge" about data expiration. Plus it knows expiration time.

Reply:
+1